sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -195,22 +197,23 @@ class FilesElement extends Polymer.Element {
 
   /**
    * Calls the ApiManager to get the list of files at the current path, and
-   * updates the fileList property.
+   * updates the _fileList property.
    * @param throwOnError whether to throw an exception if the refresh fails. This
    *                     is false by default because throwing is currently not used.
    */
-  _fetchFileList(throwOnError = false) {
+  _fetchFileList(throwOnError = false): Promise<any> {
     // Don't overlap fetch requests. This can happen because we set up fetch from several sources:
     // - Initialization in the ready() event handler.
     // - Refresh mechanism called by the setInterval().
     // - User clicking refresh button.
     // - Files page gaining focus.
     if (this._fetching) {
-      return;
+      return Promise.resolve();
```

Comment:
Seems like this should be returning the promise that is currently in flight. Otherwise, the second caller will believe the file list has been fetched and will continue processing, with a null value for the file list.